### PR TITLE
Added comments for doxygen document, and removed undefinition function

### PR DESCRIPTION
**Pull request by kim6515516**
# modified files:

vdev.h vdev.c
# remove function :

int register_vdev(void); in include/vdev.h 
Definition of this function is not exist.

### CODE REVIEW

**Comment by kesl** [comment]
Can one of the admins verify this patch?

**Comment by Wonseok** [comment]
Please modify your title.
We don't want to read long title, please make it short and complete sentence as possible as you can. In this case, "Added comments for virtual device" is enough. If you want to notice all the details, they should be located in commit.

**Comment by Wonseok** [comment]
Your commit title doesn't have full information what you did. You might want to write description for commit as following: 
Added comments as doxygen style into vdev.[ch].
Removed unused function or unnecessary code.

File: hypervisor/include/vdev.h [hunk 1]
```diff
@@ -26,10 +26,26 @@ struct vdev_info {
     unsigned int size;
     vdev_callback_t handler;
 };
-
-void vdev_init(void);
+/**
+ * @brief Initializes virtual device mapping table.
```
Comment (Wonseok) [inline hypervisor/include/vdev.h:30]: Initializes a mapping table for virtual device.
You have to add comment return type if it has no return type such as void.

File: hypervisor/include/vdev.h [hunk 2]
```diff
@@ -26,10 +26,26 @@ struct vdev_info {
     unsigned int size;
     vdev_callback_t handler;
 };
-
-void vdev_init(void);
```
Comment (Wonseok) [inline hypervisor/include/vdev.h:30]: Why did you remove vdev_init(void)? you might want to remove register_vdev() instead of vdev_dev().

File: hypervisor/include/vdev.h [hunk 3]
```diff
@@ -26,10 +26,26 @@ struct vdev_info {
...
+/**
+ * @brief Initializes virtual device mapping table.
+ */
 int register_vdev(void);
+/**
+ * @brief Registers a new virtual device.
+ * @param new_vdev information of new virtual device.
```
Comment (Wonseok) [inline hypervisor/include/vdev.h:35]: Use uppercase letter of sentence. s/information/Information

File: hypervisor/include/vdev.h [hunk 4]
```diff
@@ -26,10 +26,26 @@ struct vdev_info {
...
+ * @return When new virtual device registered successful, it returns success.\
```
Comment (Wonseok) [inline hypervisor/include/vdev.h:36]: s/When new virtual device registered successful, it returns success./It returns success when new virtual device is registered.

File: hypervisor/include/vdev.h [hunk 5]
```diff
@@ -26,10 +26,26 @@ struct vdev_info {
...
+ * otherwise it returns failed.
```
Comment (Wonseok) [inline hypervisor/include/vdev.h:37]: You don't use lowercase in this case because previous sentence is ended with period.

Comment (Wonseok) [inline hypervisor/include/vdev.h:37]: Please fix the return comment as following:
It returns success only when new virtual device is registered.
Could changed 'success' in previous sentence to MACRO that we defined?

File: hypervisor/include/vdev.h [hunk 6]
```diff
@@ -26,10 +26,26 @@ struct vdev_info {
...
+ */
 hvmm_status_t vdev_reg_device(struct vdev_info *new_vdev);
+/**
+ * @brief Emulates virtual divice .
```
Comment (Wonseok) [inline hypervisor/include/vdev.h:41]: s/Emulates virtual divice ./Emulates a virtual device.

File: hypervisor/vdev/vdev.c [hunk 7]
```diff
@@ -18,7 +18,14 @@ void vdev_init(void)
         vdev_list[i].handler = 0x0;
     }
 }
-
+/**
+ * @brief Registers a new virtual device.
+ * <br> This function added a new virtual device including device name,\
```
Comment (Wonseok) [inline hypervisor/vdev/vdev.c:23]: This function registers a new virtual device which includes device name, base address, size and handler.

File: hypervisor/vdev/vdev.c [hunk 8]
```diff
@@ -18,7 +18,14 @@ void vdev_init(void)
...
+ *  base address, size, and handler.
+ * @param new_vdev information of new virtual device.
+ * @return When new virtual device registered successful,\
```
Comment (Wonseok) [inline hypervisor/vdev/vdev.c:26]: Remove '\' in every source file. If you want to use this, please let me know why we need '\'.

File: hypervisor/vdev/vdev.c [hunk 9]
```diff
@@ -18,7 +18,14 @@ void vdev_init(void)
         vdev_list[i].handler = 0x0;
     }
 }
-
+/**
+ * @brief Registers a new virtual device.
+ * <br> This function added a new virtual device including device name,\
+ *  base address, size, and handler.
+ * @param new_vdev information of new virtual device.
```
Comment (Wonseok) [inline hypervisor/vdev/vdev.c:25]: Use uppercase letter at first character of a sentence.

File: hypervisor/vdev/vdev.c [hunk 10]
```diff
@@ -43,7 +50,23 @@ hvmm_status_t vdev_reg_device(struct vdev_info *new_vdev)
     HVMM_TRACE_EXIT();
     return result;
 }
-
+/**
+ * @brief Emulates virtual divice.
+ * <br> When virtual guest reach virtual device and hypervisor has\
```
Comment (Wonseok) [inline hypervisor/vdev/vdev.c:55]: Wrong terminology. There is no exist virtual guest is virtualization. Please use "guest OS" or "virtual machine" instead of virtual guest.

Comment (Wonseok) [inline hypervisor/vdev/vdev.c:55]: You might want to write a description such as 
Hypervisor have to trap and emulate the devices when guest OS want to access to devices.
In this time, we provide virtual devices for guest OSes with handling page fault. 
Our virtual devices works with four parameters as following steps:
1. Reads/Checks the fault address(FIPA) from HPFAR.
Emulate virtual device, if the address is registered in hypervisor as a virtual device.
Do not emulate virtual device, if the address is not registered in hypervisor as a virtual device.
2. Reads WNR to find the operation types whether the operation is store or load.
3. Reads value of access_size to find the size of the access attempted by the faulted operation.
4. Reads SRT to find the source/destination register.
